src/views/payment-sheet-views/card-view.js

```diff
@@ -571,9 +577,14 @@ CardView.prototype._hideUnsupportedCardIcons = function () {
     var cardIcon;
     var configurationCardType = constants.configurationCardTypes[paymentMethodCardType];
 
-    if (supportedCardTypes.indexOf(configurationCardType) === -1) {
-      cardIcon = this.getElementById(paymentMethodCardType + '-card-icon');
-      classList.add(cardIcon, 'braintree-hidden');
+    // We perform a check here to see if the merchant is configured for
+    // any of the cards that use the generic logo to avoid the generic
+    // logo from beind displayed in the card sheet header
+    if (cardsThatUseGenericCardLogo.indexOf(configurationCardType) > -1) {
+      if (supportedCardTypes.indexOf(configurationCardType) === -1) {
+        cardIcon = this.getElementById(paymentMethodCardType + '-card-icon');
+        classList.add(cardIcon, 'braintree-hidden');
+      }
     }
```

Comment:
We should also add test coverage around this.

Reply:
After grokking through the tests, it looks like we suppress the UnionPay icon from appearing on the card sheet header even if the merchant is configured for it by adding the `braintree-hidden` class to the card icon's HTML. I added HTML for these cards and hardcoded that class so that they don't appear in the header instead of having this check in place.